R/FilteredDataset.R

```diff
@@ -314,9 +314,38 @@ FilteredDataset <- R6::R6Class( # nolint
     #' @description
     #' Gets variable names for the filtering.
     #'
+    #' It takes the intersection of the column names
+    #' of the data and `private$.filterable_varnames` if
+    #' `private$.filterable_varnames` has positive length
+    #'
     #' @return (`character` vector) of variable names
     get_filterable_varnames = function() {
-      get_filterable_varnames(self$get_dataset())
+      varnames <- self$get_supported_filter_varnames()
+      if (length(private$.filterable_varnames) > 0) {
+        return(intersect(varnames, private$.filterable_varnames))
+      }
+      return(varnames)
+    },
+
+    #' Gets the variables which could be filtered
+    #'
+    #' This does not take into account the value
+    #' of `private$.filterable_varnames`
+    #' @return (`character` vector) of variable names
+    get_supported_filter_varnames = function() {
+      get_supported_filter_varnames(self$get_dataset())
+    },
+
+    # setters ------
+    #' @description
+    #' Set the allowed filterable variables
+    #' @param varnames (`character` or `NULL`) The variables which can be filtered
+    #' See `self$get_filterable_varnames` for more details
+    #' @return invisibly this `FilteredDataset`
```

Comment:
I can see that you do intersect on get and no input validation. I think it's a good decision given the _connectors_. Would be good to document that.